include/mull/AST/ASTSearchTask.h

```diff
@@ -0,0 +1,33 @@
+#pragma once
+
+#include "mull/Bitcode.h"
+#include "mull/Diagnostics/Diagnostics.h"
+#include "mull/Filters/FilePathFilter.h"
+#include "mull/JunkDetection/CXX/ASTStorage.h"
+
+#include <unordered_map>
+#include <vector>
+
+namespace mull {
+
+class progress_counter;
+
+class ASTSearchTask {
+public:
+  using In = const std::vector<std::unique_ptr<mull::Bitcode>>;
+  using Out = std::vector<std::pair<SourceFilePath, SingleASTUnitMutations>>;
+  using iterator = In::const_iterator;
+
+  ASTSearchTask(mull::Diagnostics &diagnostics, mull::ASTStorage &astStorage,
+                mull::MutatorKindSet mutatorKindSet, mull::FilePathFilter &pathFilter);
+
+  void operator()(iterator begin, iterator end, Out &storage, mull::progress_counter &counter);
+
+private:
+  mull::Diagnostics &diagnostics;
+  mull::ASTStorage &astStorage;
+  mull::MutatorKindSet mutatorKindSet;
```

Comment:
Just a reference should be enough here, right?

Reply:
If you look at how this mutator kind set is created in driver-cxx.cpp you will see that it is a set of mutator kinds that is created on stack with `if (tool::EnableAST` branch. Within the current implementation we could make it a reference but I think it should either be like it is now or a && move.

I expected a review comment where you would suggest me to move the AST search part to the driver so there it would make even more sense. If we don't move it to the driver I can make it a reference. Please advise.

Reply:
Actually I have changed it to references for now. It is done.